modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/util/CommerceAccountHelperImpl.java

```diff
@@ -303,34 +303,44 @@ public AccountEntry getCurrentAccountEntry(
 				if (accountEntry == null) {
 					User user = _userLocalService.getUser(userId);
 
-					ServiceContext serviceContext = new ServiceContext();
-
-					serviceContext.setCompanyId(user.getCompanyId());
-					serviceContext.setUserId(userId);
-
-					accountEntry = _accountEntryLocalService.addAccountEntry(
-						userId, AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT,
-						user.getFullName(), null, null, user.getEmailAddress(),
-						null, StringPool.BLANK,
-						AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON,
-						WorkflowConstants.STATUS_APPROVED, serviceContext);
-
-					addAccountEntryUserRel(
-						accountEntry.getAccountEntryId(), userId,
-						serviceContext);
+					if (user.equals(
+							_userLocalService.getGuestUser(
+								commerceChannel.getCompanyId()))) {
+
+						accountEntry =
+							_accountEntryLocalService.getGuestAccountEntry(
+								commerceChannel.getCompanyId());
+
+						setCurrentCommerceAccount(
+							httpServletRequest, commerceChannelGroupId,
+							AccountConstants.ACCOUNT_ENTRY_ID_GUEST);
```

Comment:
Hey @victorcdp why you added the getter of the guest account if you don't use it on line 316?

Reply:
Hi @alessiorendina, what do you mean by getter of the guest account?
If you mean the getGuestUSer on line 307, I put that in because we need to compare the current user to the guestUser.
If you mean the getGustAccountEntry on 311, I put that in because there's already an AccountEntry for the guest user created, so we need to return that one if the current user is a guest.

And there's no specific reason i didn't use the accountEntry ID on line 316, from my knowledge, both the ACCOUNT_ENTRY_ID_GUEST constant and the accountEntry.getAccountEntryId() should be fine, since the Guest AccountEntry is created using the constant, but if you want me to change it for consistency, just say the word.

And about the reordering, moving the code doesn't necessarily fix the bug, the real fix is the if block starting on line 306. Before the fix, the code would reach this point and would try to add an AccountEntry for the Guest User, but would throw an exception because the Guest User has no name (and as i said, there's already an AccountEntry for the guest user).
I just reordered because previously, the `if (accountEntry == null) {` block's purpose was to call `setCurrentCommerceAccount` passing the Id for the Guest AccountEntry, but the code was never reached since the exception was thrown and the Guest AccountEntry was never retrieved. With this fix though, this if block is superfluous because the if block i put in checks for the same thing, so i just combined the two blocks.

Reply:
Ok, I understood now. Then there is no need to call the userLocalService to check if the users are the same, you can just use `user.isGuestUser()`. Also, there is no need to set the current account in the if-else block. Once the block ends, you can just set the current account like this:

`setCurrentCommerceAccount(httpServletRequest, commerceChannelGroupId, accountEntry.getAccountEntryId());`

Reply:
Hi @alessiorendina, i made some tests and the original if block (on line 359) is actually necessary if the site isn't B2C or B2X, otherwise all the minium pages would break. So i just left the way it was, but i still implemented your other request (user.isGuestUser). Can you see if it's alright now? Thank you.